IDinsight/aaq-core
# [After #238] UD Validation #1 - Local run

Reviewer: @suzinyou 
Estimate: 1hr

---

## Ticket

Fixes: [AAQ-570](https://idinsight.atlassian.net/browse/AAQ-570)

## Description

### Goal

Run validation locally. Returns results like this:

{
  "true_positives": 93,
  "false_negatives": 4,
  "true_negatives": 135,
  "false_positives": 713,
  "precision": 0.11538461538461539,
  "recall": 0.9587628865979382
}

More reporting coming in future PRs!

### Changes


### Future Tasks (optional)
- Run using GA + AWS Secrets + Data in S3
- Run for different thresholds and return ROC/PR-curve

## How has this been tested?

You can run it locally.

1. Run llm-proxy: `make setup-llm-proxy`
2. Copy files from S3 to your local machine. 
      - S3 URI: s3://aaq-static-resources/validation_data/momconnect_2022/
      - Files: mc_urgency_*.csv
3. Update Makefile in `core_backend/validation/urgency_detection` with paths to your file. Note that paths should be relative to root of repo i.e. `aaq_core`
4. Run `make ud-validation`

NOTE: Suggest starting with the `*_small.csv` files to avoid long runs.

## To-do before merge (optional)

#238 should be merged

## Checklist

Fill with `x` for completed. 


(Delete any items below that are not relevant)


[AAQ-570]: https://idinsight.atlassian.net/browse/AAQ-570?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
